### DISCUSSION TITLE
Unable to access global variables such as dbdict, _ED, inside my custom script

### DISCUSSION THREAD

**Discussion by Sheharyar570** [TPC-Council/HammerDB#740] [Q&A] [answered]
Hi HammerDB Team,

I'm running a custom script to run and I'm unable to access the global variables that are set according to the db configuration i.e. dbdict, bm, _ED. I've checked that these variables are available in vurun and in vucreate as well and in customscript proc. But once the VUs start running the script, I'm unable to access those variables inside the script.

Is this behavior intentional? How can I access the global variables specifically configurations etc inside my custom script?

Any guidance would be appreciated!
Thanks

**Comment by sm-shaw** [comment, marked as answer]
Yes, that is correct and intentional. The Virtual Users are threads and can create and manage their own variables. If you want to share variables between threads you need to use thread shared variables https://www.tcl.tk/man/tcl8.6/ThreadCmd/tsv.htm,  https://www.activestate.com/blog/threads-done-right-tcl/

**Reply by Sheharyar570** [reply]
Thanks a lot for the reply, I'll give it a try.
